TellCore/NativeMethods.cs

```diff
@@ -24,29 +194,29 @@ internal static class NativeMethods
         [return: MarshalAs(UnmanagedType.CustomMarshaler, MarshalCookie = "out", MarshalTypeRef = typeof(TelldusUtf8Marshaler))]
         internal static extern string tdGetModel(int deviceId);
 
-        [DllImport("TelldusCore.dll")]
+        [DllImport("TelldusCore.dll", BestFitMapping = false, ThrowOnUnmappableChar = true)]
         [return: MarshalAs(UnmanagedType.CustomMarshaler, MarshalCookie = "out", MarshalTypeRef = typeof(TelldusUtf8Marshaler))]
         internal static extern string tdGetDeviceParameter(
             int deviceId,
             [MarshalAs(UnmanagedType.CustomMarshaler, MarshalCookie = "in", MarshalTypeRef = typeof(TelldusUtf8Marshaler))] string name,
             [MarshalAs(UnmanagedType.CustomMarshaler, MarshalCookie = "in", MarshalTypeRef = typeof(TelldusUtf8Marshaler))] string defaultValue);
 
-        [DllImport("TelldusCore.dll")]
+        [DllImport("TelldusCore.dll", BestFitMapping = false, ThrowOnUnmappableChar = true)]
         internal static extern bool tdSetName(
             int deviceId,
             [MarshalAs(UnmanagedType.CustomMarshaler, MarshalCookie = "in", MarshalTypeRef = typeof(TelldusUtf8Marshaler))] string name);
 
-        [DllImport("TelldusCore.dll")]
+        [DllImport("TelldusCore.dll", BestFitMapping = false, ThrowOnUnmappableChar = true)]
         internal static extern bool tdSetProtocol(
             int deviceId,
             [MarshalAs(UnmanagedType.CustomMarshaler, MarshalCookie = "in", MarshalTypeRef = typeof(TelldusUtf8Marshaler))] string protocol);
 
-        [DllImport("TelldusCore.dll")]
+        [DllImport("TelldusCore.dll", BestFitMapping = false, ThrowOnUnmappableChar = true)]
         internal static extern bool tdSetModel(
             int deviceId,
             [MarshalAs(UnmanagedType.CustomMarshaler, MarshalCookie = "in", MarshalTypeRef = typeof(TelldusUtf8Marshaler))] string model);
 
-        [DllImport("TelldusCore.dll")]
+        [DllImport("TelldusCore.dll", BestFitMapping = false, ThrowOnUnmappableChar = true)]
```

Comment:
Are `BestFirMapping` and `ThrowOnUnmappableChar` attempts to bail out earlier if something goes wrong (unmappable char or the like) or what would be the purpose of this?